intel-types/src/lib.rs

```diff
@@ -52,12 +54,267 @@ impl<T: Copy + Debug + PartialEq + BitAnd<Output = T>> PartialEq<T> for Masked<T
-///
-/// Section 38.9, Table 38-7
-#[derive(Debug)]
-#[repr(C, align(4096))]
-pub struct XSave([[u64; 32]; 16]);
+/// This type represents an MMX register, used to perform operations on 64-bit packed integer data.
+/// It includes the 10-byte value as well as 6 bytes of padding.
+#[derive(Debug, Default)]
+#[repr(C)]
+pub struct Mm(u128);
+
+/// This type represents an XMM 128-bit data register, used to operate on packed single-
+/// precision floating-point operands.
+#[derive(Debug, Default)]
+#[repr(C)]
+pub struct Xmm(u128);
+
+bitflags! {
+    /// x87 Floating Point Unit (FPU) Control Word
+    /// Section 8.1.5
+    pub struct Fcw: u16 {
+
+        /// Invalid Operation
+        const INV_OP = 1 << 0;
+
+        /// Denormal Operand
+        const DENORM_OP = 1 << 1;
+
+        /// Zero Divide
+        const ZERO_DIV = 1 << 2;
+
+        /// Overflow
+        const OVERFLOW = 1 << 3;
+
+        /// Underflow
+        const UNDERFLOW = 1 << 4;
+
+        /// Precision
+        const PREC = 1 << 5;
+
+        /// Precision Control 0
+        const PREC_CTRL0 = 1 << 8;
+
+        /// Precision Control 1
+        const PREC_CTRL1 = 1 << 9;
+
+        /// Rounding Control 0
+        const ROUND_CTRL0 = 1 << 10;
+
+        /// Rounding Control 1
+        const ROUND_CTRL1 = 1 << 11;
+
+        /// Infinity Control
+        const INFINITY_CTRL = 1 << 12;
+    }
+}
+
+bitflags! {
+
+    /// 32-bit register providing status and control bits used in SIMD floating-point operations
+    pub struct MxCsr: u32 {
+
+        /// Invalid Operation Flag
+        const INV_OP = 1 << 0;
+
+        /// Denormal Flag
+        const DENORM = 1 << 1;
+
+        /// Divide by Zero Flag
+        const ZERO_DIV = 1 << 2;
+
+        /// Overflow Flag
+        const OVERFLOW = 1 << 3;
+
+        /// Underflow Flag
+        const UNDERFLOW = 1 << 4;
+
+        /// Precision Flag
+        const PREC = 1 << 5;
+
+        /// Denormals are Zeros
+        const DENORM_ARE_ZEROS = 1 << 6;
+
+        /// Invalid Operation Mask
+        const INV_OP_MASK = 1 << 7;
+
+        /// Denormal Operation Mask
+        const DENORM_MASK = 1 << 8;
+
+        /// Divide by Zero Mask
+        const ZERO_DIV_MASK = 1 << 9;
+
+        /// Overflow Mask
+        const OVERFLOW_MASK = 1 << 10;
+
+        /// Underflow Mask
+        const UNDERFLOW_MASK = 1 << 11;
+
+        /// Precision Mask
+        const PREC_MASK = 1 << 12;
+
+        /// Rounding Control 0
+        const ROUND_CTRL0 = 1 << 13;
+
+        /// Rounding Control 1
+        const ROUND_CTRL1 = 1 << 14;
+
+        /// Flush to Zero
+        const FLUSH_TO_ZERO = 1 << 15;
+    }
+}
+
+bitflags! {
+    /// Section 13.4.3
+    pub struct XcompBv: u64 {
+
+        /// Compacted form is used for the layout of the XSAVE EXtended Region
+        const COMPACT = 1 << 63;
+    }
+}
+
+/// The legacy region of an XSAVE area comprises the 512 bytes starting at the area's base address.
+/// See Table 13-1.
+#[repr(C)]
```

Comment:
Is there an alignment requirement?

Reply:
The documentation says it takes up the first 512 bytes of the `XSave` region, starting from the `XSave` base address ([13.4.1](https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-vol-1-manual.pdf)). I understood this as a size requirement, but would this also be considered an alignment requirement? Other structs in the Intel documentation explicitly say "must be 512-byte aligned," etc.

Reply:
No. It will say aligned if they want alignment.